File: week_8_and_9/3_Ruby/fizz_buzz/my_solution.rb
# U3.W8-9: 


# I worked on this challenge [by myself, with: ].

# 2. Pseudocode
#Iterate through the array and replace th enumber with each necessary condition.



# 3. Initial Solution

def super_fizzbuzz(array)
	new_array = []
	array.each do |num|
		if num % 15 == 0
			new_array << 'FizzBuzz'
		elsif num % 5 == 0 
			new_array << 'Buzz'
		elsif num % 3 == 0 
			new_array << 'Fizz'
		else 
			new_array << num 
		end 
	end 
	return new_array
end



# 4. Refactored Solution






# 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

puts super_fizzbuzz([3]) == ["Fizz"]
puts super_fizzbuzz([5]) == ["Buzz"]
puts super_fizzbuzz([15]) == ["FizzBuzz"]
puts super_fizzbuzz([1, 2, 3]) == [1, 2, "Fizz"]
puts super_fizzbuzz([15, 5, 3, 1]) == ["FizzBuzz", "Buzz", "Fizz", 1]



# 5. Reflection 
#Have done several examples of this excersize, so was pretty strait forward

File: week_8_and_9/2_JavaScript/translate_to_js/my_solution.js
// U3.W8-9: Challenge you're converting


// I worked on this challenge [by myself, with: ].

// 2. Pseudocode
//def super_fizzbuzz(array)
//	new_array = []
//	array.each do |num|
//		if num % 15 == 0
//			new_array << 'FizzBuzz'
//		elsif num % 5 == 0 
//			new_array << 'Buzz'
//		elsif num % 3 == 0 
//			new_array << 'Fizz'
//		else 
//			new_array << num 
//		end 
//	end 
//	return new_array
//end

//translate this 

// 3. Initial Solution

var fizz_buzz = function (array) {
	var new_array = [];
	for (var i = 0; i < array.length; i++) {
		if (array[i] % 15 == 0) {
			new_array.push("FizzBuzz");
		}
		else if (array[i] % 5 == 0) {
			new_array.push("Buzz");
		}
		else if (array[i] % 3 == 0) {
			new_array.push("Fizz");
		}
		else {
			new_array.push(array[i]);
		} 
	}
	return new_array 
}

// 4. Refactored Solution






// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

 console.log(fizz_buzz([15, 5, 3, 4])==['FizzBuzz', 'Buzz', 'Fizz', 4])




// 5. Reflection 